src/components/PopupWithForm.js

```diff
@@ -17,11 +37,18 @@ export default class PopupWithForm extends Popup {
     return inputvalues;
   }
 
+  setInputs(inputValues) {
+    this.inputs.forEach((input) => {
+      input.value = inputValues[input.name];
+    });
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this.form.addEventListener("submit", (e) => {
       e.preventDefault();
-      this.submit(e, this._getInputValues());
+      this.renderLoading(true);
```

Comment:
Needs correction: you should start the loading right before the api request in index.js as the form submission could be handled using different ways.